src/core/net/nat64_translator.cpp

```diff
@@ -128,15 +131,16 @@ Translator::Result Translator::TranslateFromIp6(Message &aMessage)
         ExitNow(res = kDrop);
     }
 
-    if (!ip6Header.GetDestination().MatchesPrefix(mNat64Prefix))
+    if (!ip6Header.GetDestinationAddress().MatchesPrefix(mNat64Prefix))
     {
         ExitNow(res = kNotTranslated);
     }
 
-    mapping = FindOrAllocateMapping(ip6Header.GetSource());
+    mapping = FindOrAllocateMapping(ip6Header, srcPortOrId);
```

Comment:
`srcPortOrId` is set to zero at the top but is not updated afterward until this call. Will `FindOrAllocateMapping()` set it? 

In `TranslateToIp6()`, we have the following:

```cpp
#if OPENTHREAD_CONFIG_NAT64_PORT_TRANSLATION_ENABLE
dstPortOrId = mapping->mSrcPortOrId;
#endif
```

Do we need something similar here? So it can updated based on the `mapping` we find?
Maybe `mapping-> mTranslatedPortOrId`?

Reply:
Yes, I addressed this in next comment.